### ISSUE TITLE
Raw disk is identified as part of pool instead of the partitions that belong to it

### ISSUE THREAD

**Issue by wgtthompson** [zfsonlinux/zfs#2876]
##### Summary
Automatic detection of the correct device(s) that belong to a pool fails when the (correct) device is a disk partition (instead of the whole disk).  The result shows as follows on Debian 7.7 Wheezy:

```
# zpool import
   pool: pool
     id: 16149256128135082299
  state: FAULTED
 status: One or more devices contains corrupted data.
 action: The pool cannot be imported due to damaged devices or data.
   see: http://zfsonlinux.org/msg/ZFS-8000-5E
 config:

        pool                                                        FAULTED  corrupted data
          mirror-0                                                  ONLINE
            ata-VMware_Virtual_IDE_Hard_Drive_00000000000000000001  UNAVAIL  corrupted data
            ata-VMware_Virtual_IDE_Hard_Drive_01000000000000000001  UNAVAIL  corrupted data
```
The above example is on VMWare but similar occurs without virtualization (observed with the same set of disks).

I raised this problem in zfs-discuss at https://groups.google.com/a/zfsonlinux.org/d/topic/zfs-discuss/3sryjOnTnXU/discussion

##### Version
Here are the package versions for the non-virtualized machine

```
# dpkg -l | grep zfs
ii  dkms                                  2.2.0.3-1.2+zfs6              all          Dynamic Kernel Module Support Framework
ii  libzfs2                               0.6.3-1~wheezy                amd64        Native ZFS filesystem library for Linux
ii  zfs-dkms                              0.6.3-1~wheezy                all          Native ZFS filesystem kernel modules for Linux
ii  zfsonlinux                            3~wheezy                      all          archive.zfsonlinux.org trust package
ii  zfsutils                              0.6.3-1~wheezy                amd64        command-line tools to manage ZFS filesystems
```

##### Work-around
Create a directory that contains soft-links to the partitions that actually belong to the ZFS pool.  Import using `zpool import -d /path/to/softlinks <poolname>`.  Add this line to `/etc/init.d/zfs-mount` for each pool so that the pools are brought up on boot.

**Comment by Finkregh** [comment]
why is this not closed, when this is a dup

**Comment by behlendorf** [comment]
@Finkregh Yes, this can be closed as a duplicate.  For anyone hitting this issue I've proposed a fix in #3219.  It would be very helpful to have someone independently verify the fix.
